### ISSUE TITLE
‘Implement a C API in JavaScript’ ，the implement must be in window global environment?

### ISSUE THREAD

**Issue by yangfangfang1204** [emscripten-core/emscripten#17242]
First, In C code,  define the interface 
`extern "C"{
extern void test(int a,int b);

int callAdd ()
{
  test(2,3);
}
}`

Second, in library.js ,implement the interface in JavaScript:
`
test__proxy:'async',
test__sig:'vii;,
test:function(a,b)
{
console.log(a+", and"+b);
JS_Test();
}
`
Then , compile , `emcc test.cpp -o test.js --js-library library.js`

Last in html/js , i call the function 'callAdd' , so my question : the `JS_Test()` must in window global environment to implement it ?

**Comment by sbc100** [comment]
Yes, `JS_Test` must be defines somewhere, either on the global object (or it could also be in the module scope if you use MODULARIZE).
